test/unit/services/sshAdapter_spec.js

```diff
@@ -0,0 +1,357 @@
+"use strict";
+
+var AWS        = require("aws-sdk");
+var sinon      = require("sinon");
+var expect     = require("chai").expect;
+var SshAdapter = require("../../../lib/services/sshAdapter");
+var Bluebird   = require("bluebird");
+var options    = {};
+var ec2        = new AWS.EC2();
+
+Bluebird.promisifyAll(ec2);
+options.ec2 =  ec2;
+
+describe("The SSH Adapter Class", function () {
+	var id                   = "i-fbf47257";
+	var SSH_POLLING_INTERVAL = 30; //Time in Seconds
+
+	describe("Can Ssh", function () {
+		var instanceStatusStub;
+		var result;
+		var sshAdapter = new SshAdapter(ec2);
+		before(function () {
+			instanceStatusStub = sinon.stub(ec2,"describeInstanceStatusAsync", function () {
+				var data = {
+					InstanceStatuses : [ {
+						SystemStatus : {
+							Details : [ {
+								Status : "passed"
+							} ]
+						}
+					} ]
+				};
+				return (Bluebird.resolve(data));
+			});
+			sshAdapter.canSsh(id)
+			.then(function (data) {
+				result = data;
+			})
+			.catch(function (err) {
+				result = err;
+			});
+		});
+
+		after(function () {
+			instanceStatusStub.restore();
+		});
+
+		it("is able to connect", function () {
+			expect(instanceStatusStub.called).to.be.true;
+			expect(instanceStatusStub.args[ 0 ][ 0 ].InstanceIds[ 0 ]).to.equal(id);
+			expect(result).to.be.true;
+		});
+	});
+
+	describe("Tries to ssh", function () {
+		describe("as soon as instance is created", function () {
```

Comment:
I don't understand these descriptions. Ideally, these descriptions will align with each other so that when a test case fails it reads almost like a user story. So something like "The SSH adapter when trying to connect to ssh before the instance has started fails to connect.".